test/resources/Subscription.spec.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+require('../testUtils');
+
+describe('Subscription Resource', function () {
+
+    var account, subscription, rand;
+
+    before(function (done) {
+        rand = new Date().getTime();
+        var a = {
+            externalKey: rand,
+            name: 'subscription ' + rand,
+            currency: 'USD'
+        };
+        K.Account.create(a, 'mocha', 'testing', 'testing', function (error, result) {
+            account = result;
+            done();
+        });
+    });
+
+    describe('create', function () {
+        it('should create a subscription', function (done) {
+            var s = {
+                accountId: account.accountId,
+                externalKey: 'sub-' + rand,
+                productName: 'Standard',
+                productCategory: 'BASE',
+                priceList: 'DEFAULT',
+                billingPeriod: 'MONTHLY'
+            };
+            K.Subscription.create(s, 'mocha', 'testing', 'testing', function (error, result) {
+                if (error)
+                    throw error;
+
+                expect(result.accountId).to.equal(account.accountId);
+                expect(result.billingPeriod).to.equal(s.billingPeriod);
+                expect(result.externalKey).to.equal(s.externalKey);
+                expect(result.state).to.equal('ACTIVE');
+                assert.isNotNull(result.bundleId);
+                assert.isNotNull(result.subscriptionId);
+                assert.isArray(result.events);
+                subscription = result;
+                done();
+            });
+        });
+
+        //TODO default catalog doesn't have any addons - could upload catalog as part of test?
```

Comment:
Indeed, that's why for tests, we usually use the `SpyCarAdvanced.xml` catalog instead of `SpyCarBasic.xml`. But :+1: for uploading a catalog instead (we do something similar in the [Ruby integration tests](https://github.com/killbill/killbill-integration-tests/tree/master/killbill-integration-tests/resources), where we have specific catalogs testing various features).